app/battle/preparation.py

```diff
@@ -0,0 +1,23 @@
+def apply_armour(knight_data: dict) -> dict:
+    knight_data["protection"] = 0

```

Comment:
Consider initializing 'protection' only if it doesn't already exist in 'knight_data'. This will prevent overwriting existing protection values if the function is called multiple times.